Fix initializing the database for the first time

ungeskriptet opened the pull request:
The if-else condition was always returning true even though existing_ids was empty. Add an additional check to see if we even have data in the database

ungeskriptet committed:
CreateCollection: Check if existing_ids is set

This fixes initializing the database for the first time

ungeskriptet committed:
README: Fix typo

ungeskriptet committed:
ArchiveYouTube: Let yt-dlp choose the best format

Fixes: 4909d90dab70 ("Let youtube-dl select the best format")

ungeskriptet committed:
ArchiveYouTube: Show upload progress when logs are enabled

Alyetama commented on internetarchive_youtube/archive_youtube.py:
```diff
@@ -415,7 +415,7 @@ def process_video(self, video: dict, mongodb: bool, jsonbin: bool,
 
         if not video['uploaded']:
-            resp = self.upload(video, md, identifier, fname)
+            resp = self.upload(video, md, identifier, fname, not self.no_logs)
```
upload is a static method, so `self` will raise an error. Also, I don't understand the reason behind negating the user selection of `no_logs`.

ungeskriptet replied:
Even when logs are enabled it's not possible to see the upload progress and speed. Passing the no_logs boolean here makes it's possible to show the upload speed and progress from the internetarchive library. As for the error, I don't get any.

Alyetama commented on internetarchive_youtube/archive_youtube.py:
```diff
@@ -292,7 +292,7 @@ def upload(video: dict, md: dict, identifier: str,
         r = None
         try:
-            r = upload(identifier, files=[fname], metadata=md)
+            r = upload(identifier, files=[fname], metadata=md, verbose=verbose)
```
I don't understand using `verbose` when `verbose` is not used anywhere in the method `upload`.

Alyetama commented on internetarchive_youtube/archive_youtube.py:
```diff
@@ -263,7 +263,7 @@ def download(self, video: dict, ydl_opts: dict,
     @staticmethod
     def upload(video: dict, md: dict, identifier: str,
-               fname: str) -> Optional[int]:
+               fname: str, verbose: bool) -> Optional[int]:
```
Is there a reason to add this argument when it's not referenced anywhere in the function?

ungeskriptet replied:
Well I'm not a Python export but it's being referenced in line 295 and so on?

Alyetama commented:
Thank you for initiating this pull request! I’ve added some comments on one file for your review. I’ll either test the additional commits or wait until the PR is no longer in draft status.

ungeskriptet commented:
> I’ve added some comments on one file for your review.

Hi, the PR is now ready. Did you perhaps click on "Start a review" when adding your comments instead of "Add single comment"? I'm not able to see your review comments

ungeskriptet committed:
main: Add __main__.py

ungeskriptet committed:
Add support for cookies

Add support for cookies by checking if the 'COOKIES' environment
variable is set.
